libyul/YulStack.h

```diff
@@ -172,6 +172,11 @@ class YulStack: public langutil::CharStreamProvider
 	solidity::frontend::OptimiserSettings m_optimiserSettings;
 	langutil::DebugInfoSelection m_debugInfoSelection{};
 
+	/// Provider of the Solidity sources that the Yul code was generated from.
+	/// Necessary when code snippets are requested as a part of debug info. When null, code snippets are omitted.
+	/// NOTE: Not owned by YulStack, the user must ensure that it is not destoroyed before the stack is.
+	langutil::CharStreamProvider const* m_soliditySourceProvider{};
```

Comment:
I think that the best solution is to just store a pointer to the source provider (i.e. `CompilerStack` in `YulStack`. This will require us to worry about its lifetime, but on the other hand we'll no longer have any issues with things being printed differently due to debug info.

Reply:
I don't think that the lifetime is a big worry in this case, so fully agree!